docs/pages/build-reference/variables.mdx

```diff
@@ -49,6 +49,15 @@ You can access these variables in your application using the techniques describe
 
 See the [eas.json reference](/build/eas-json) for more information.
 
+### Setting environment variables dynamically
+
+You can also set environment variables dynamically during the build process. The `set-env` executable is available in the `PATH` on EAS Build workers, and can be used to set environment variables that will be visible in the next build phases.
+
+For example, you can add the following in one of the [EAS Build hooks](/build-reference/npm-hooks/) and the environment variable `EXAMPLE_ENV` will be available until the end of the build job.
+
+```bash
+set-env EXAMPLE_ENV "example value"
```

Comment:
```suggestion
<Terminal cmd={[ 'set-env EXAMPLE_ENV "example value"']} />
```

Let's use the `Terminal` component instead of a `bash` code snippet to represent the action to be done in the terminal or represent a command.